### PR TITLE
Add TLH git commit attribution

### PR DESCRIPTION
**Pull request by diegopetrucci**
## Summary
- Add boolean-only TLH commit attribution settings with /toggle-tlh-git-attribution.
- Prompt enabled sessions to include the TLH footer on agent-created git commits.
- Block obvious unattributed bash-tool git commits without rewriting commands or changing git push behavior.
- Document the default footer and disable/re-enable path.

## Validation
- npm run validate

## Summary by CodeRabbit

* **New Features**
  * TLH can append a branded commit footer to agent-created git commits and will block obvious inline git commits missing that footer.
  * Added a persistent /toggle-tlh-git-attribution command to enable/disable this per-profile; isolated-profile defaults and unset/true vs false semantics are documented.

* **Documentation**
  * README and changelog updated with usage, setting semantics, and the exact footer template.

* **Tests**
  * Expanded tests covering attribution resolution, parsing edge cases, runtime enforcement (child/primary), and toggle persistence/backup behavior.

### CODE REVIEW

File: extensions/the-last-harness/primary-agent-runtime.ts [hunk 1]
```diff
@@ -541,6 +548,11 @@ function createTlhPrimaryAgentRuntime(
 		});
 
 		pi.on("tool_call", async (event, ctx) => {
+			const commitAttributionState = resolveTlhCommitAttribution(getTlhGlobalSettings(ctx.cwd).tlh?.attribution);
+			if (event.toolName === "bash") {
+				const reason = getTlhGitCommitAttributionBlockReason(event.input.command, commitAttributionState);
+				return reason ? { block: true, reason } : undefined;
```
[the remark this answers is not in the document]
Comment (diegopetrucci) [reply extensions/the-last-harness/primary-agent-runtime.ts:574]: Fixed in 582aab3. Child-mode startup now registers a child-specific before_agent_start/tool_call runtime so delegated sessions still get the TLH attribution prompt and bash commit guard without re-enabling parent-only controls. Added runtime and static coverage for the PI_SUBAGENT_CHILD path.

File: extensions/the-last-harness/attribution.ts [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+	if (current) {
+		tokens.push(current);
+	}
+	return tokens;
+}
+
+function getGitCommitArguments(segment: string): string[] | undefined {
+	const tokens = tokenizeShellWords(segment);
+	if (tokens[0]?.toLowerCase() !== "git") {
+		return undefined;
```
[the remark this answers is not in the document]
Comment (diegopetrucci) [reply extensions/the-last-harness/attribution.ts]: Fixed in 582aab3. The bash attribution guard now unwraps simple shell control-flow prefixes (`then`, `else`, `do`, and leading `!`) before parsing each segment, so wrapped inline `git commit` commands are detected. Added attribution/runtime tests for wrapped commits and wrapped here-doc commits.

File: extensions/the-last-harness/attribution.ts [hunk 3]
```diff
@@ -0,0 +1,610 @@
...
+	let searchOffset = 0;
+	for (const line of footer.split("\n").filter(Boolean)) {
+		const index = command.indexOf(line, searchOffset);
+		if (index === -1) {
+			return false;
+		}
+		searchOffset = index + line.length;
+	}
+	return true;
+}
```
[the remark this answers is not in the document]
Comment (diegopetrucci) [reply extensions/the-last-harness/attribution.ts]: Fixed in effe18e. The process-substitution path now derives the effective commit message content and requires the TLH footer to be the exact trailing suffix of that rendered message, so trailing extra text is rejected. This handling is intentionally conservative/fail-closed for more complex shell/process-substitution forms: if we cannot prove the final rendered message ends with the exact footer, the guard blocks the commit.

File: extensions/the-last-harness/attribution.ts [hunk 4]
```diff
@@ -0,0 +1,610 @@
...
+			continue;
+		}
+		if (token.startsWith("-C") || token.startsWith("-c")) {
+			continue;
+		}
+		if (["--config-env", "--git-dir", "--namespace", "--super-prefix", "--work-tree"].some((option) => token.startsWith(`${option}=`))) {
+			continue;
+		}
+	}
+	return undefined;
```
[the remark this answers is not in the document]
Comment (diegopetrucci) [reply extensions/the-last-harness/attribution.ts]: Fixed in effe18e. The guard now strips leading shell wrappers/prefixes for obvious inline forms (`if`/`then`/`else`/`do`, `command`, `!`, and leading env assignments) before parsing `git commit`, so wrapped commits like `if git commit ...; then`, `command git commit ...`, and `FOO=bar git commit ...` are checked and blocked unless the exact TLH footer is present. Added helper, runtime, and static coverage for these wrapped forms.
